lib/ansible/module_utils/docker_common.py

```diff
@@ -541,3 +541,103 @@ def pull_image(self, name, tag="latest"):
         new_tag = self.find_image(name, tag)
 
         return new_tag, old_tag == new_tag
+
+
+def compare_dict_allow_more_present(av, bv):
+    '''
+    Compare two dictionaries for whether every entry of the first is in the second.
+    '''
+    for key, value in av.items():
+        if key not in bv:
+            return False
+        if bv[key] != value:
+            return False
+    return True
+
+
+def compare_generic(a, b, method, type):
```

Comment:
I recognize that this will be covered by the existing integration tests, but would it be worthwhile to add unit tests around this functionality? There's enough logic in here that I think it merits additional validation.

Reply:
A very good point! I'll add something.